Sources/SKTestSupport/TestJSONRPCConnection.swift

```diff
@@ -92,6 +92,7 @@ public final class TestClient: LanguageServerEndpoint {
   var oneShotNotificationHandlers: [((Any) -> Void)] = []
 
   public var allowUnexpectedNotification: Bool = false
+  public var allowUnexpectedRequest: Bool = false
```

Comment:
I had this to prevent ApplyEdit from crashing, but I ended up replicating the notification behavior to requests as well to be able to retrieve the client's ApplyEdit response - let me know if you think something different works better!

Reply:
Where is it `allowUnexpectedRequest  == true`?

Reply:
Ended up being a leftover from before, will remove it so it can be added when it's really needed.